applications/GeoMechanicsApplication/custom_conditions/thermal_point_flux_condition.h

```diff
@@ -0,0 +1,68 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//
+//  Main authors:    Mohamed Nabi
+//                   John van Esch
+//
+
+#pragma once
+
+#include "custom_conditions/T_condition.h"
+#include "includes/serializer.h"
+
+namespace Kratos {
+
+template <unsigned int TDim, unsigned int TNumNodes>
+class KRATOS_API(GEO_MECHANICS_APPLICATION) GeoThermalPointFluxCondition
+    : public GeoTCondition<TDim, TNumNodes> {
+public:
+    using GeometryType   = Geometry<Node>;
+    using PropertiesType = Properties;
+    using NodesArrayType = GeometryType::PointsArrayType;
+
+    KRATOS_CLASS_INTRUSIVE_POINTER_DEFINITION(GeoThermalPointFluxCondition);
+
+    GeoThermalPointFluxCondition();
+
+    GeoThermalPointFluxCondition(IndexType NewId, GeometryType::Pointer pGeometry);
+
+    GeoThermalPointFluxCondition(IndexType NewId,
+                            GeometryType::Pointer pGeometry,
+                            PropertiesType::Pointer pProperties);
+
+    ~GeoThermalPointFluxCondition() override;
+
+    Condition::Pointer Create(IndexType NewId,
+                              NodesArrayType const& rThisNodes,
+                              PropertiesType::Pointer pProperties) const override
+    {
+        return Kratos::make_intrusive<GeoThermalPointFluxCondition>(
+            NewId, this->GetGeometry().Create(rThisNodes), pProperties);
+    }
+
+protected:
+    void CalculateRHS(Vector& rRightHandSideVector,
+                      const ProcessInfo& CurrentProcessInfo) override;
+
+private:
+    friend class Serializer;
+
+    void save(Serializer& rSerializer) const override
+    {
+        KRATOS_SERIALIZE_SAVE_BASE_CLASS(rSerializer, Condition)
+    }
+
+    void load(Serializer& rSerializer) override
+    {
+        KRATOS_SERIALIZE_LOAD_BASE_CLASS(rSerializer, Condition)
```

Comment:
And the same thing here:
```suggestion
        using BaseType = GeoTCondition<TDim, TNumNodes>;
        KRATOS_SERIALIZE_LOAD_BASE_CLASS(rSerializer, BaseType)
```
To avoid repeating the `BaseType` alias, you mag also consider to put it near the other aliases at the top of the class definition.